nldb/core.py

```diff
@@ -65,7 +64,7 @@ def cache_chat_completion(prompt_messages):
     if os.path.exists(cache_file):
         return json.loads(open(cache_file).read())
     # otherwise, run the completion
-    resp = openai.ChatCompletion.create(
+    resp = await openai.ChatCompletion.acreate(
         model="gpt-3.5-turbo",
         messages=prompt_messages,
         temperature=0,
```

Comment:
This change affects `.env` support:

```
  File "[...]/nldb/nldb/api.py", line 26, in ask
    sql_statement = await nldb.text_to_sql(q)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/nldb/nldb/core.py", line 102, in text_to_sql
    resp = await cache_chat_completion(prompt_messages)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/nldb/nldb/core.py", line 67, in cache_chat_completion
    resp = await openai.ChatCompletion.acreate(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/.venv/lib/python3.11/site-packages/openai/api_resources/chat_completion.py", line 45, in acreate
    return await super().acreate(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/.venv/lib/python3.11/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 214, in acreate
    ) = cls.__prepare_create_request(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/.venv/lib/python3.11/site-packages/openai/api_resources/abstract/engine_api_resource.py", line 106, in __prepare_create_request
    requestor = api_requestor.APIRequestor(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/.venv/lib/python3.11/site-packages/openai/api_requestor.py", line 134, in __init__
    self.api_key = key or util.default_api_key()
                          ^^^^^^^^^^^^^^^^^^^^^^
  File "[...]/.venv/lib/python3.11/site-packages/openai/util.py", line 186, in default_api_key
    raise openai.error.AuthenticationError(
openai.error.AuthenticationError: No API key provided. You can set your API key in code using 'openai.api_key = <API-KEY>', or you can set the environment variable OPENAI_API_KEY=<API-KEY>). If your API key is stored in a file, you can point the openai module at it with 'openai.api_key_path = <PATH>'. You can generate API keys in the OpenAI web interface. See https://platform.openai.com/account/api-keys for details.
```

suggestion:
```suggestion
        temperature=0,
        api_key=OPENAPI_KEY,
```

with the appropriate import

Reply:
interesting - async calls don't have the same access to env vars?

Reply:
I could not find anything on this, I'm afraid

Reply:
@tomdyson https://github.com/tomdyson/nldb/pull/4/files#diff-e8afac852bf9e2abbd3e90920b1d69e287ccd44885e38f8d99057a3cf746bd61R14 needs `, OPENAPI_KEY`